msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenRequestExecutor.java

```diff
@@ -67,6 +67,15 @@ OAuthHttpRequest createOauthHttpRequest() throws SerializeException, MalformedUR
             params.put("claims", Collections.singletonList(claimsRequest));
         }
 
+        if(msalRequest.requestContext().apiParameters().extraQueryParameters() != null ){
+            for(String key: msalRequest.requestContext().apiParameters().extraQueryParameters().keySet()){
+                    if(params.containsKey(key)){
```

Comment:
There is no need to add this validation. If app developer sets this and overrides some params that MSAL computes, it's their problem. 

Consider adding just a `log.warning`

Reply:
Sure I was going by the requirements, thinking this is what all the MSALs do. https://github.com/AzureAD/microsoft-authentication-library-for-java/issues/256 Point 5.

Reply:
Updated in the latest commit.